amcl/src/amcl_node.cpp

```diff
@@ -477,6 +497,37 @@ AmclNode::AmclNode() :
   laser_check_interval_ = ros::Duration(15.0);
   check_laser_timer_ = nh_.createTimer(laser_check_interval_,
                                        boost::bind(&AmclNode::checkLaserReceived, this, _1));
+
+  // publish ready signal
+  ready_pub_.publish(true);
+
+  //start action server
+  set_inital_pose_action_.start();
+
+  driver_pose_pub_ = nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>(srs::ChuckTopics::internal::ODOMETRY_INITIAL_POSE, 2, true);
+}
+
+void AmclNode::executeInitialPoseCB(const move_base_msgs::SetInitialPoseGoalConstPtr &goal)
+{
+  ROS_INFO("Executing, inital pose server action");
+  if(use_emulator_)
+  {
+    driver_pose_pub_.publish(goal->initialPose);
+    ROS_INFO("Sleeping for brainstem to publish new tf/odometry");
+    ros::Duration(1.0).sleep();
```

Comment:
magic sleep numbers? this seems like possible dangerous async behavior. what happens if we didnt sleep? maybe this should be a polling action with a timeout instead?

Reply:
yeah so first thing is that this sleep is needed only in sim/emulator mode. Basically amcl needs to have updated odometery from brainstem. and if we don't sleep the amcl position doesn't get updated. there are other places in code where we do this as well. probably should update all of that as well

Reply:
this is how much we wait when we use `2d position` in RVIz